app/notifiers/course/announcement_notifier.rb

```diff
@@ -1,8 +1,11 @@
 class Course::AnnouncementNotifier < Notifier::Base
   # To be called when an announcement is made.
   def new_announcement(user, announcement)
-    create_activity(actor: user, object: announcement, event: :new).
-      notify(announcement.course, :email).
-      save
+    email_enabled = Course::Settings::AnnouncementsComponent.
+                    email_enabled?(announcement.course, :new_announcement)
+
+    activity = create_activity(actor: user, object: announcement, event: :new)
+    activity.notify(announcement.course, :email) if email_enabled
+    activity.save
```

Comment:
Do we want to create the activity even when the notification is disabled ?  To me, we should not.